SnM/SnM_Marker.cpp

```diff
@@ -345,3 +345,228 @@ void InsertMarker(COMMAND_T* _ct)
 	UpdateTimeline();
 	Undo_OnStateChangeEx2(NULL, SWS_CMD_SHORTNAME(_ct), UNDO_STATE_MISCCFG, -1);
 }
+
+TempoMarker::TempoMarker(double _dTimePos, int _measurePos, double _dBeatPos, double _dBpm, int _timesig_num, int _timesig_denom, bool _bLinearTempo, RawTempoMarkerData _rawData)
+{
+	m_dTimePos = _dTimePos;
+	m_measurePos = _measurePos;
+	m_dBeatPos = _dBeatPos;
+	m_dBpm = _dBpm;
+	m_timesig_num = _timesig_num;
+	m_timesig_denom = _timesig_denom;
+	m_bLinearTempo = _bLinearTempo;
+	m_rawData = _rawData;
+}
+
+bool GetAllTempoMarkers(WDL_PtrList<void>* _tempoMarkers, ReaProject* _proj)
+{
+	if (_tempoMarkers)
+	{
+		_tempoMarkers->Empty();
+		double dTimePosOut;
+		int measurePosOut;
+		double dBeatPosOut;
+		double dBpmOut;
+		int timesig_numOut;
+		int timesig_denomOut;
+		bool bLinearTempoOut;
+
+		map<int, RawTempoMarkerData> tempoDataMap;
+
+		MediaTrack* masterTrack = GetMasterTrack(_proj);
+		TrackEnvelope* tempoEnvEx = GetTrackEnvelope(masterTrack, 0);
```

Comment:
This may not give the tempo envelope if there are other envelopes on the master track.

```
GetTrackEnvelopeByName(masterTrack, "Tempo map")
```